network-store-iidm-impl/src/main/java/com/powsybl/network/store/iidm/impl/PhaseTapChangerImpl.java

```diff
@@ -91,10 +94,8 @@ public int getStepCount() {
     }
 
     @Override
-    public PhaseTapChangerStep getStep(int tapPosition) {
-        var attributes = getAttributes();
-        int tapPositionIndex = tapPosition - attributes.getLowTapPosition();
-        return new PhaseTapChangerStepImpl(this, tapPositionIndex);
+    protected PhaseTapChangerStepImpl createStep(int tapPosition) {
+        return new PhaseTapChangerStepImpl(this, tapPosition);
```

Comment:
I would keep this method here and only factorize the `getTapPositionIndex` part
```suggestion
    public PhaseTapChangerStep getStep(int tapPosition) {
        int tapPositionIndex = getTapPositionIndex(tapPosition);
        return new PhaseTapChangerStepImpl(this, tapPositionIndex);
```